### ISSUE TITLE
What if I select 100 threads?

### ISSUE THREAD

**Issue by mertduver** [thelinuxchoice/instashell#45]
hello everyone I have a question, 
I select 100 threads and it worked really fast but iam not sure that is it worked properly? and how can i sure? what the deal with this treahd thing, the program says select <20 but why?
what happens if I select 100? will it miss some passwords?

**Comment by silver0321** [comment]
@mertduver You see that threads are like isolated tasks that will be done by the processors in your computer. You must have heard terms like Quad Core CPU, Dual Core CPU and so on. Quad means four, Dual means two and so on. 

Now threads are isolated entities which have isolated lifetime within the process which uses the thread itself. **You can picture the flow of program execution as a thread**.

Each line in the program gets executed from top to bottom, just like you would go through a thread in the physical world. Imagine if you have multiple threads with the same instruction or different one, and you have to go through these threads reading some sort of instruction and you have to do this simultaneously on all these given thread, you would have a lot of options but what the **CPU does is that it cycles through these so called threads and executes a large number of instruction from a single thread in the pool of threads per clock speed or more, you kind of achieve a illusion that all threads are executed at the same time but the CPU is so fast it cycles through all the threads.**

**TL;DR**

Each CPU can execute any number of threads but more number of threads can slow the execution of a single thread because the CPU is just cycling through the threads. So **having a large number of threads will not be any faster in theory but if the tool was faster then your processor maybe kind of have a different thing going on(latest tech I'm not aware of)**. But the more threads you have, the more difficulty you give for the **os scheduler**.

**So giving 1 thread for each CPU would be the ideal thread count for any program.**

You can see have much CPU you have by executing the following command in linux,

```
$ nproc
```

Or if you Quad core then 4 thread count would be ideal.

*Learn more here:* https://www.youtube.com/watch?v=7ENFeb-J75k

**Comment by gjrahul** [comment]
> hello everyone I have a question,
> I select 100 threads and it worked really fast but iam not sure that is it worked properly? and how can i sure? what the deal with this treahd thing, the program says select <20 but why?
> what happens if I select 100? will it miss some passwords?

That would great an heavy stress on the instagram server, you might even get caught if 100 threads are used. 

100 threads means 100 password / sec
